shortfin/README.md

```diff
@@ -111,10 +111,13 @@ Run platform independent tests only:
 pytest tests/
 ```
 
-Run tests including for a specific platform:
+Run tests including for a specific platform (in this example, a gfx1100 AMDGPU):
+
+(note that not all tests are system aware yet and some may only run on the CPU)
 
 ```bash
-pytest tests/ --system amdgpu
+pytest tests/ --system amdgpu \
+    --compile-flags="--iree-hal-target-backends=rocm --iree-hip-target=gfx1100"
```

Comment:
Clever - passing a _list_ of flags through via this new option.

We have a few tests in other places that pull `--iree-hip-target` from a `IREE_HIP_TARGET` environment variable but that needs the tests to know about the env var for that specific backend and doesn't allow for passing extra flags.

One recent example is https://github.com/nod-ai/SHARK-Platform/pull/373, with this code:
```python
CPU_SETTINGS = {
    "device_flags": [
        "-iree-hal-target-backends=llvm-cpu",
        "--iree-llvmcpu-target-cpu=host",
    ],
    "device": "local-task",
}
IREE_HIP_TARGET = os.environ.get("IREE_HIP_TARGET", "gfx1100")
gpu_settings = {
    "device_flags": [
        "-iree-hal-target-backends=rocm",
        f"--iree-hip-target={IREE_HIP_TARGET}",
    ],
    "device": "hip",
}
```

fyi @stbaione we might want to use this `--compile-flags` pattern for pytest options across more of the project, including that integration test.

One advantage to putting the flags _in the test_ vs loading them from CLI flags is that the test itself can encode xfail behavior for each set of parameters. Hopefully we won't have that many xfails in shortfin 🤞 

In yet another place, I have config files specify compile and run flags along with lists of xfails that are used to instantiate parameterized tests, like https://github.com/iree-org/iree/blob/main/tests/external/iree-test-suites/onnx_ops/onnx_ops_cpu_llvm_sync.json

Reply:
This will get better. I'm going to poke through enough API so that the compile flags can be completely inferred. But was too much for tonight (and in any case, you do want to be able to set them explicitly too).

Reply:
Presumably then, too, if worse comes to worse, XFAIL could be inferred based on what device it actually is.

Reply:
I don't anticipate having a lot of XFAIL'y stuff in shortfin. We should just be fixing things.